browser/ui/webui/brave_rewards_page_ui.cc

```diff
@@ -494,26 +488,38 @@ void RewardsDOMHandler::Init() {
   rewards_service_ =
       brave_rewards::RewardsServiceFactory::GetForProfile(profile);
   rewards_service_->StartProcess(base::DoNothing());
+
+  ads_service_ = brave_ads::AdsServiceFactory::GetForProfile(profile);
+}
+
+void RewardsDOMHandler::IsInitialized(
+    const base::ListValue* args) {
+  AllowJavascript();
+
+  if (rewards_service_ && rewards_service_->IsInitialized()) {
+    CallJavascriptFunction(
+        "brave_rewards.initialized",
+        base::Value(0));
+  }
+}
+
+void RewardsDOMHandler::OnJavascriptAllowed() {
   if (rewards_service_) {
     rewards_service_->AddObserver(this);
   }
 
-  ads_service_ = brave_ads::AdsServiceFactory::GetForProfile(profile);
   if (ads_service_) {
     ads_service_->AddObserver(this);
   }
 }
 
-void RewardsDOMHandler::IsInitialized(
-    const base::ListValue* args) {
-  if (!web_ui()->CanCallJavascript()) {
-    return;
+void RewardsDOMHandler::OnJavascriptDisallowed() {
```

Comment:
you also need to call `weak_factory_.InvalidateWeakPtrs()` to make sure no callbacks complete after this is called